### PR TITLE
First draft Maas2013rectifier_2 sritank

### PR DESCRIPTION
**Pull request by sritank**

### CODE REVIEW

File: _summaries/maas2013rectifier_2.md [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
-author: # Your GitHub id
-score: # How did you like this paper 0(dislike) to 10(love)
+author: sritank
+score: 8/10
 ---
 
 TODO: Summarize the paper:
 * What is the core idea?
+
+DNNs with ReLU and other non-sigmoid activation functions performed better on image recognition tasks. So the paper explores using DNNs as acoustic models and how rectifier non-linearities can improve estimation accuracy of the acoustic model.
```
Comment (saikm200022) [inline _summaries/maas2013rectifier_2.md:13]: The second sentence is fine, although I would suggest removing the "So" and say "This paper" as the "So" makes the summary seem a bit informal . I also think you should mention something about some of the core terms they used to explain the ReLu DNN performance over the sigmoidal DNN performance (namely sparsity and dispersion) as this is something that is emphasized in the paper.

File: _summaries/maas2013rectifier_2.md [hunk 2]
```diff
@@ -1,19 +1,36 @@
...
+
 * How is it realized (technically)?
+
+The authors train a DNN acoustic model in an LVCSR (large vocabulary continuous speech recognition) task. To study the impact of different non-linearities on the training accuracy, they do not use regularization or pre-trained weights for the network. 
+
+For the same speech recognition task, they replace the GMM/HMM acoustic model with a DNN. The DNN training inputs are MFCC (Mel frequency cepstral coefficients) features of the audio signal. The output of the DNN is the senone likelihood probability function. They train DNNs (2-4 layers containing 2048 units/layer) with sigmoid, ReLU and Leaky ReLU activation functions using stochastic gradient descent.
```
Comment (saikm200022) [inline _summaries/maas2013rectifier_2.md:19]: I think it is a good idea to mention the loss function that is used (Cross Entropy) and the batch size of 256. I would also mention that they tried different learning rates in the set [0.1, 0.05, 0.01, 0.005, 0.001]. I found it interesting that they started with a momentum of 0.5 and increased it 0.9 after a certain number of SGD iterations, which might be worth adding to your summary as well.

File: _summaries/maas2013rectifier_2.md (hunk 1 above)
Comment (saikm200022) [inline _summaries/maas2013rectifier_2.md:13]: Just for full clarity I recommend starting the first sentence with "Deep Neural Networks (DNNs)" to avoid any confusion on that acronym.

File: _summaries/maas2013rectifier_2.md [hunk 3]
```diff
@@ -1,19 +1,36 @@
...
+
 * How is it realized (technically)?
+
+The authors train a DNN acoustic model in an LVCSR (large vocabulary continuous speech recognition) task. To study the impact of different non-linearities on the training accuracy, they do not use regularization or pre-trained weights for the network. 
+
+For the same speech recognition task, they replace the GMM/HMM acoustic model with a DNN. The DNN training inputs are MFCC (Mel frequency cepstral coefficients) features of the audio signal. The output of the DNN is the senone likelihood probability function. They train DNNs (2-4 layers containing 2048 units/layer) with sigmoid, ReLU and Leaky ReLU activation functions using stochastic gradient descent. 
+
 * How well does the paper perform?
+
+The paper evaluated the WER and cross entropy (for frame-wise error) obtained by replacing the GMM (Gaussian Mixture Model) acoustic model with the different DNNs. They trained the DNN with a 300 Hour switchboard phone conversation dataset (LDC97S62).
```
Comment (saikm200022) [inline _summaries/maas2013rectifier_2.md:23]: I would actually write out WER (word error rate) since in the paper they write it out and then present the acronym.

Comment (saikm200022) [inline _summaries/maas2013rectifier_2.md:23]: "They trained the DNN with a 300 Hour switchboard phone conversation dataset (LDC97S62). " - This sentence probably makes more sense to be placed in the "How is it realized (technically)?" section since it talks about training details.

File: _summaries/maas2013rectifier_2.md [hunk 4]
```diff
@@ -1,19 +1,36 @@
...
+
 * How well does the paper perform?
+
+The paper evaluated the WER and cross entropy (for frame-wise error) obtained by replacing the GMM (Gaussian Mixture Model) acoustic model with the different DNNs. They trained the DNN with a 300 Hour switchboard phone conversation dataset (LDC97S62). 
+
+They found that ReLU DNNs had a lower WER and cross entropy indicating a better estimation of the senone likelihood function. They observed a 2% decrease in WER when using ReLU DNNs compared to ones using sigmoidal activations. Even 4 layer Sigmoidal DNNs performed slightly worse than 2 layer ReLU DNNs despite having 80% more free parameters.
+
+They attribute this better performance to the lower activation probability in a ReLU DNN. They show that the ReLU Layers are more dispersed (low standard deviation) making it a sparser representation than the sigmoidal DNN.
```
Comment (saikm200022) [inline _summaries/maas2013rectifier_2.md:27]: It might be a good idea to define activation probability (although you could also just define during your presentation).

Comment (saikm200022) [reply _summaries/maas2013rectifier_2.md:27]: I recommend adding a sentence talking about how dispersion is beneficial because it distributes information uniformly across hidden units.

File: _summaries/maas2013rectifier_2.md [hunk 5]
```diff
@@ -1,19 +1,36 @@
...
+
+For the same speech recognition task, they replace the GMM/HMM acoustic model with a DNN. The DNN training inputs are MFCC (Mel frequency cepstral coefficients) features of the audio signal. The output of the DNN is the senone likelihood probability function. They train DNNs (2-4 layers containing 2048 units/layer) with sigmoid, ReLU and Leaky ReLU activation functions using stochastic gradient descent. 
+
 * How well does the paper perform?
+
+The paper evaluated the WER and cross entropy (for frame-wise error) obtained by replacing the GMM (Gaussian Mixture Model) acoustic model with the different DNNs. They trained the DNN with a 300 Hour switchboard phone conversation dataset (LDC97S62). 
+
+They found that ReLU DNNs had a lower WER and cross entropy indicating a better estimation of the senone likelihood function. They observed a 2% decrease in WER when using ReLU DNNs compared to ones using sigmoidal activations. Even 4 layer Sigmoidal DNNs performed slightly worse than 2 layer ReLU DNNs despite having 80% more free parameters.
```
Comment (saikm200022) [inline _summaries/maas2013rectifier_2.md:25]: It might be a good idea to mention how ReLu DNNs are not as susceptible to the vanishing gradient problem.

**Review by saikm200022 — CHANGES_REQUESTED** [review]
Nice summary! I have some suggestions that I think will make your summary even better.

File: _summaries/maas2013rectifier_2.md (hunk 1 above)
Comment (sritank) [reply _summaries/maas2013rectifier_2.md:13]: done

File: _summaries/maas2013rectifier_2.md (hunk 3 above)
Comment (sritank) [reply _summaries/maas2013rectifier_2.md:23]: done

File: _summaries/maas2013rectifier_2.md (hunk 1 above)
Comment (sritank) [reply _summaries/maas2013rectifier_2.md:13]: Done

File: _summaries/maas2013rectifier_2.md (hunk 3 above)
Comment (sritank) [reply _summaries/maas2013rectifier_2.md:23]: I thought the same initially but in the paper it comes under the Experiments section and they talk about their network performance on that dataset. So I thought it would be more appropriate in the paper performance question

File: _summaries/maas2013rectifier_2.md (hunk 5 above)
Comment (sritank) [reply _summaries/maas2013rectifier_2.md:25]: I added it, but to clarify, the authors took that into account by initialization of the network weights. I don't think they mentioned any saturation during their training phase.

File: _summaries/maas2013rectifier_2.md (hunk 2 above)
Comment (sritank) [reply _summaries/maas2013rectifier_2.md:19]: I added the point about the loss function. Regarding the others, they never mention if the learning rates or momentum increased performance in any way. The network would probably just take more iterations to converge if momentum was constant. But the performance of the final trained network is unchanged. So I don't think it's necessary to mention about the learning rate and momentum

File: _summaries/maas2013rectifier_2.md (hunk 4 above)
Comment (sritank) [reply _summaries/maas2013rectifier_2.md:27]: I was going to define it during the presentation (didn't want to write definitions in a summary). 
Added plots from paper.

File: _summaries/maas2013rectifier_2.md (hunk 3 above)
Comment (saikm200022) [reply _summaries/maas2013rectifier_2.md]: That makes sense

File: _summaries/maas2013rectifier_2.md (hunk 5 above)
Comment (saikm200022) [reply _summaries/maas2013rectifier_2.md]: "Each time we add a hidden layer, rectifier DNNs show a greater absolute reduction in WER than sigmoidal DNNs. We believe this effect results from the lack of vanishing gradients in rectifier networks" - They mentioned this statement in the paper, so I think it is worth mentioning this finding they observed.

File: _summaries/maas2013rectifier_2.md (hunk 2 above)
Comment (saikm200022) [reply _summaries/maas2013rectifier_2.md]: They mentioned that 0.01 led to the fastest convergence, but I could see how that is not crucial to the summary. On second thought, stating the momentum is likely not useful here either.

File: _summaries/maas2013rectifier_2.md (hunk 5 above)
Comment (sritank) [reply _summaries/maas2013rectifier_2.md]: Great point. I added it to the review

**Review by saikm200022 — APPROVED** [review]
Draft Looks Good!

**Review by saikm200022 — APPROVED** [review]
